src/main/java/seedu/address/logic/commands/DeleteListingCommand.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.listing.Listing;
+import seedu.address.model.person.Name;
+
+/**
+ * Deletes a listing identified by its name.
+ */
+public class DeleteListingCommand extends Command {
+
+    public static final String COMMAND_WORD = "deleteListing";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the listing specified by its name.\n"
+            + "Parameters: LISTING_NAME\n"
+            + "Example: " + COMMAND_WORD + " n/Warton House";
+
+    public static final String MESSAGE_DELETE_LISTING_SUCCESS = "Successfully deleted listing: %1$s";
+    public static final String MESSAGE_LISTING_NOT_FOUND = "This listing does not exist in EZSTATE";
+
+    private final Name targetName;
+
+    public DeleteListingCommand(Name targetName) {
+        this.targetName = targetName;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Listing listingToDelete = model.getListingByName(targetName);
```

Comment:
I think using a .contains() method that returns whether a `listing` exits in the `model's` `listings` may be appropriate instead of using a `null` check. To see how to implement this, you can see the seedu.address.logic.commands.DeleteClientProfileCommand.java class for more information.